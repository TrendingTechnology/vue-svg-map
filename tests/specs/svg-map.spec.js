import { shallowMount } from '@vue/test-utils'
import map from '../mocks/fake-map'
import SvgMap from '../../src/components/svg-map'

describe('SvgMap component', () => {
	it('renders map with default props', () => {
		const wrapper = shallowMount(SvgMap, {
			propsData: { map },
		})

		expect(wrapper.element).toMatchSnapshot()
	})

	it('renders map with custom props', () => {
		const wrapper = shallowMount(SvgMap, {
			propsData: {
				map,
				locationClass: 'locationClass',
				locationTabindex: 'locationTabindex',
				locationRole: 'locationRole',
				isLocationSelected: () => 'isLocationSelected()',
			},
		})

		expect(wrapper.element).toMatchSnapshot()
	})
})
