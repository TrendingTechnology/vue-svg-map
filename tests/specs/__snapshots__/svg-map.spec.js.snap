// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgMap component renders map with custom props 1`] = `
<svg
  aria-label="label"
  class="svg-map"
  viewBox="viewBox"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    aria-checked="isLocationSelected()"
    aria-label="name0"
    class="svg-map__location locationClass"
    d="path0"
    id="id0"
    name="name0"
    role="locationRole"
    tabindex="locationTabindex"
  />
  <path
    aria-checked="isLocationSelected()"
    aria-label="name1"
    class="svg-map__location locationClass"
    d="path1"
    id="id1"
    name="name1"
    role="locationRole"
    tabindex="locationTabindex"
  />
  <path
    aria-checked="isLocationSelected()"
    aria-label="name2"
    class="svg-map__location locationClass"
    d="path2"
    id="id2"
    name="name2"
    role="locationRole"
    tabindex="locationTabindex"
  />
</svg>
`;

exports[`SvgMap component renders map with default props 1`] = `
<svg
  aria-label="label"
  class="svg-map"
  viewBox="viewBox"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    aria-label="name0"
    class="svg-map__location"
    d="path0"
    id="id0"
    name="name0"
  />
  <path
    aria-label="name1"
    class="svg-map__location"
    d="path1"
    id="id1"
    name="name1"
  />
  <path
    aria-label="name2"
    class="svg-map__location"
    d="path2"
    id="id2"
    name="name2"
  />
</svg>
`;
